temporalio/runtime.py

```diff
@@ -174,15 +178,54 @@ def _to_bridge_config(self) -> temporalio.bridge.runtime.PrometheusConfig:
         )
 
 
+class MetricBuffer:
+    """A buffer that can be set on :py:class:`TelemetryConfig` to record
+    metrics instead of ignoring/exporting them.
+
+    .. warning::
+        It is very important that the buffer size is set to a high number and
+        that :py:meth:`retrieve_updates` is called regularly. Metric updates
+        will not be lost, so they can cause the system to halt if there is not
+        enough room in the buffer.
+    """
+
+    def __init__(self, buffer_size: int) -> None:
+        """Create a buffer with the given size.
+
+        .. warning::
+            It is very important that the buffer size is set to a high number
+            and is drained regularly. See :py:class:`MetricBuffer` warning.
+
+        Args:
+            buffer_size: Size of the buffer. Set this to a large value.
```

Comment:
Some guidance beyond large is useful here I think

Reply:
Suggestions? I didn't know of a good value so I intentionally didn't set a default. I don't know how frequently updates are emitted to add any more info.

Reply:
I added "A value in the tens of thousands or higher is plenty reasonable" but not sure what other guidance I can provide that I don't already with warnings at the class level and elsewhere explaining that frequent draining is required. Suggestions?

Reply:
I think that's good. 10k ish seems reasonable. It's hard to guess at.